llvm/lib/IR/RepoTicket.cpp

```diff
@@ -74,54 +74,77 @@ updateInitialDigestAndGetDependencies(const GlobalObject *GO, MD5 &GOHash,
   GOInfoMap::const_iterator Pos;
   if (const auto GV = dyn_cast<GlobalVariable>(GO)) {
     ++GONum.VarNum;
-    Pos = calculateInitialDigestAndDependencies(GV, GOIMap);
+    Pos = calculateInitialDigestAndDependenciesAndContributions(GV, GOIMap);
   } else if (const auto Fn = dyn_cast<Function>(GO)) {
     ++GONum.FuncNum;
-    Pos = calculateInitialDigestAndDependencies(Fn, GOIMap);
+    Pos = calculateInitialDigestAndDependenciesAndContributions(Fn, GOIMap);
   } else {
     llvm_unreachable("Unknown global object type!");
   }
   GOHash.update(Pos->second.InitialDigest.Bytes);
   return Pos->second.Dependencies;
 }
 
-const DependenciesType &
-updateDigestGONumAndGetDependencies(const GlobalObject *GO, MD5 &GOHash,
-                                    GOInfoMap &GOIMap, GONumber &GONum) {
+const GODigestState updateDigestGONumAndGetDependencies(const GlobalObject *GO,
+                                                        MD5 &GOHash,
+                                                        GOInfoMap &GOIMap,
+                                                        GONumber &GONum) {
   auto It = GOIMap.find(GO);
   if (It != GOIMap.end()) {
     const GOInfo &GOInformation = It->second;
     GOHash.update(GOInformation.InitialDigest.Bytes);
-    return GOInformation.Dependencies;
+    return {GOInformation.Dependencies, true};
```

Comment:
Doesn't this make a copy of its first argument (a `DependenciesType`)? Is that what you want?

Reply:
No, I have changed it to a "reference" instead of a "copy".